cmd/tcl/testworkflow-init/data/state.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2024 Testkube.
+//
+// Licensed as a Testkube Pro file under the Testkube Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//	https://github.com/kubeshop/testkube/blob/main/licenses/TCL.txt
+
+package data
+
+import (
+	"bytes"
+	"encoding/gob"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/kubeshop/testkube/pkg/tcl/expressionstcl"
+)
+
+const (
+	defaultInternalPath       = "/.tktw"
+	defaultTerminationLogPath = "/dev/termination-log"
+)
+
+type state struct {
+	Status TestWorkflowStatus
+	Steps  map[string]*StepInfo
+	Output map[string]string
+}
+
+var State = &state{
+	Steps:  map[string]*StepInfo{},
+	Output: map[string]string{},
+}
+
+func (s *state) GetStep(ref string) *StepInfo {
+	_, ok := State.Steps[ref]
+	if !ok {
+		State.Steps[ref] = &StepInfo{Ref: ref}
+	}
+	return State.Steps[ref]
+}
+
+func (s *state) GetOutput(name string) (expressionstcl.Expression, bool, error) {
+	v, ok := s.Output[name]
+	if !ok {
+		return expressionstcl.None, false, nil
+	}
+	expr, err := expressionstcl.Compile(v)
+	return expr, true, err
+}
+
+func (s *state) GetSelfStatus() string {
+	if Step.Executed {
+		return string(Step.Status)
+	}
+	v := s.GetStep(Step.Ref)
+	if v.Status != StepStatusPassed {
+		return string(v.Status)
+	}
+	return string(Step.Status)
+}
+
+func (s *state) GetStatus() string {
+	if Step.Executed {
+		return string(Step.Status)
+	}
+	if Step.InitStatus == "" {
+		return string(s.Status)
+	}
+	v, err := RefStatusExpression(Step.InitStatus)
+	if err != nil {
+		return ""
+	}
+	str, _ := v.Static().StringValue()
+	return str
+}
+
+func readState(filePath string) {
+	b, err := os.ReadFile(filePath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			panic(err.Error())
```

Comment:
do we need panic everywhere?

Reply:
Basically, these are clearly not User issues, but likely actual implementation bugs if happens, nothing that User may deal with

Reply:
Added `Failf` helper, and used in in places where error code is expected; `panic` is left in:

* places which could be used by other programs (i.e. emit)
* places where something unusual happened, likely due implementation bug (i.e. state not saved), not user error